### DISCUSSION TITLE
Mismatched SSL template

### DISCUSSION THREAD

**Discussion by pranav095** [projectdiscovery/nuclei-templates#5624] [General]
Hello, can someone explain this https://github.com/projectdiscovery/nuclei-templates/blob/master/ssl/mismatched-ssl.yaml
or send any reference for the same.

**Comment by olearycrew** [comment]
This template looks for a mis-match between the domain the SSL certificate for the domain.

For example:

`nuclei -id mismatched-ssl-certificate -u https://boleary.dev`

Doesn't return anything because the certificat for `boleary.dev` is valid for that domain.

However, using https://badssl.com/ and their "wrong host" example, we get a returned value:

`nuclei -id mismatched-ssl-certificate -u https://wrong.host.badssl.com/`

<img width="1283" alt="CleanShot 2023-04-14 at 12 11 58@2x" src="https://user-images.githubusercontent.com/6044920/232098791-e27ff571-0359-4b98-ba82-bdd7050eb62f.png">
